src/include/test/unit_vamana_metadata.cc

```diff
@@ -0,0 +1,191 @@
+      test_data_root / "nano" / "vamana" / "vamana_test_index_metadata",
+      test_data_root / "nano" / "vamana" / "vamana_test_index",
+  };
+
+  auto group_uri = test_uris[test_case];
+
+  tiledb::Context ctx;
+  tiledb::Config cfg;
+  tiledb::VFS vfs(ctx);
+  REQUIRE(vfs.is_dir(group_uri));
+  auto read_group = tiledb::Group(ctx, group_uri, TILEDB_READ, cfg);
+
+  auto x = vamana_index_metadata();
+  x.load_metadata(read_group);
+
+  for (auto& [name, value] : expected_str) {
+    tiledb_datatype_t v_type;
+    uint32_t v_num;
+    const void* v;
+    CHECK(read_group.has_metadata(name, &v_type));
+    if (!read_group.has_metadata(name, &v_type)) {
+      continue;
+    }
+
+    read_group.get_metadata(name, &v_type, &v_num, &v);
+    CHECK((v_type == TILEDB_STRING_ASCII || v_type == TILEDB_STRING_UTF8));
+    std::string tmp = std::string(static_cast<const char*>(v), v_num);
+    CHECK(!empty(value));
+    CHECK(tmp == value);
+  }
+  for (auto& [name, value] : expected_arithmetic) {
+    tiledb_datatype_t v_type;
+    uint32_t v_num;
+    const void* v;
+    CHECK(read_group.has_metadata(name, &v_type));
+    if (!read_group.has_metadata(name, &v_type)) {
+      continue;
+    }
+
+    read_group.get_metadata(name, &v_type, &v_num, &v);
+
+    if (name == "temp_size") {
+      CHECK((v_type == TILEDB_UINT64 || v_type == TILEDB_UINT32));
+      if (v_type == TILEDB_UINT64) {
+        CHECK(value == *static_cast<const uint64_t*>(v));
+      } else if (v_type == TILEDB_FLOAT64) {
+        CHECK(value == (uint64_t) * static_cast<const double*>(v));
+      }
+    }
+    CHECK(
+        (v_type == TILEDB_UINT32 || v_type == TILEDB_UINT64 ||
+         v_type == TILEDB_FLOAT64 || v_type == TILEDB_FLOAT32));
+
+    switch (v_type) {
+      case TILEDB_FLOAT64:
+        CHECK(value == *static_cast<const double*>(v));
+        break;
+      case TILEDB_FLOAT32:
+        CHECK(value == *static_cast<const float*>(v));
+        break;
+      case TILEDB_UINT64:
+        CHECK(value == *static_cast<const uint64_t*>(v));
+        break;
+      case TILEDB_UINT32:
+        CHECK(value == *static_cast<const uint32_t*>(v));
+        break;
+      case TILEDB_STRING_UTF8:
+        CHECK(name == "mystery string utf8");
+        break;
+      default:
+        CHECK(name == "mystery name");
+        break;
+    }
+  }
+}
+
+#if 0
+TEST_CASE("vamana_metadata: open group", "[vamana_metadata]") {
+  tiledb::Context ctx;
+  tiledb::Config cfg;
+
+  auto read_group = tiledb::Group(ctx, sift_group_uri, TILEDB_READ, cfg);
+  auto x = vamana_index_metadata();
+
+  SECTION("load metadata") {
+    x.load_metadata(read_group);
+  }
+
+  SECTION("load and dump metadata -- for manual inspection") {
+    x.load_metadata(read_group);
+    x.dump();
+  }
+
+  SECTION("Compare two constructed objects") {
+    x.load_metadata(read_group);
+    ivf_flat_index_metadata y;
+    y.load_metadata(read_group);
+    CHECK(x.compare_metadata(y));
+  }
+}
+#endif
```

Comment:
Can we remove `#if 0`?

Reply:
Finished changing this from an ivf index test to a vamana index test, and relocated it to a more logical place in the file.

Note that this just uses "nano" for testing.  At some point we need to have full-sized examples based on sift et al.